hs_core/hydro_realtime_signal_processor.py

```diff
@@ -25,49 +24,71 @@ def handle_save(self, sender, instance, **kwargs):
+                                             str(type(instance)), newbase.short_id)
 
 
                     # if object is private or becoming private, delete from index
                     else:  # not to be shown in discover
                         try:
                             index = self.connections[using].get_unified_index().get_index(newsender)
-                            index.remove_object(newinstance, using=using)
+                            index.remove_object(newbase, using=using)
                         except NotHandled:
-                            logger.exception("Failure: delete of %s with short_id %s failed.", 
-                                             str(type(instance)), newinstance.short_id)
+                            logger.exception("Failure: delete of %s with short_id %s failed.",
+                                             str(type(instance)), newbase.short_id)
 
         elif isinstance(instance, ResourceAccess):
-            # automatically a BaseResource; just call the routine on it. 
-            newinstance = instance.resource
-            newsender = BaseResource
-            self.handle_save(newsender, newinstance)
+            # automatically a BaseResource; just call the routine on it.
+            try:
+                newbase = instance.resource
+                self.handle_save(BaseResource, newbase)
+            except Exception as e:
+                logger.exception("{} exception: {}".format(type(instance), e))
 
+        elif isinstance(instance, CoreMetaData):
+            try:
+                newbase = instance.resource
+                self.handle_save(BaseResource, newbase)
+            except Exception:
+                logger.exception("{} exception: {}".format(type(instance), e))
+
+        elif isinstance(instance, AbstractMetaDataElement):
+            try:
+                # resolve the BaseResource corresponding to the metadata element.
+                newbase = instance.metadata.resource
+                self.handle_save(BaseResource, newbase)
+            except Exception as e:
+                logger.exception("{} exception: {}".format(type(instance), e))
+
+        elif isinstance(instance, HStoreField):
+            try:
+                newbase = BaseResource.objects.get(extra_metadata=instance)
+                self.handle_save(BaseResource, newbase)
+            except Exception as e:
+                logger.exception("{} exception: {}".format(type(instance), e))
 
     def handle_delete(self, sender, instance, **kwargs):
         """
-        Given an individual model instance, determine which backends the
-        delete should be sent to & delete the object on those backends.
-        This never worked well. Just set the resource to not discoverable 
-        before deleting it. This will delete it from SOLR properly. 
+        Ignore delete events as this is accomplished separately. 
         """
         pass
```

Comment:
Aren't these changes already in #4173?

Reply:
Yes. I simply copied the file here because it won't work without them.